x-pack/plugins/notifications/server/routes/api/v1/notifications/notify.js

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import Joi from 'joi';
+import { wrap } from 'boom';
+
+/**
+ * Check the incoming request parameters to see if the action should be allowed to fire.
+ *
+ * @param {Object|null} action The action selected by the user.
+ * @param {String} actionId The ID of the requested action from the user.
+ * @param {Object} data The incoming data from the user.
+ * @returns {Object|null} The error object, or null if no error.
+ */
+export function checkForErrors(action, actionId, data) {
+  if (action === null) {
+    return {
+      message: `Unrecognized action: '${actionId}'.`,
+    };
+  } else {
+    let validLicense = false;
+
+    try {
+      validLicense = action.isLicenseValid();
+    } catch (e) {
+      // validLicense === false
+    }
+
+    if (validLicense === false) {
+      return {
+        message: `Unable to perform '${action.name}' action due to the current license.`,
+      };
+    }
+  }
+
+  const fields = action.getMissingFields(data);
+
+  if (fields.length !== 0) {
+    return {
+      message: `Unable to perform '${action.name}' action due to missing required fields.`,
+      fields
+    };
+  }
+
+  return null;
+}
+
+export async function sendNotification(server, notificationService, actionId, data, reply, { _checkForErrors = checkForErrors } = { }) {
```

Comment:
Is exposing _checkForErrors only because of tests?  I think there is a way, perhaps if you put it into it's own file, that you can mock it so you don't have to expose it here just for testing purposes.

Reply:
Yeah, I am not used to Jest yet, so I did not honestly consider mocking the import. I'll swap that around.

Reply:
I ended up not changing this and instead documented the parameter's purpose (for tests). It's handy to control that function instead of having to mock all of the behaviors.

Reply:
Which ... now that I typed it and reread it, I could do by mocking the import.